src/jsCallback.php

```diff
@@ -188,4 +175,46 @@ public function getURL($mode = 'callback')
     {
         throw new Exception('Do not use getURL on jsCallback, use getJSURL()');
     }
+
+    /**
+     * Transform response into proper js Action and return it.
+     *
+     * @param $response
+     *
+     * @return jsExpression|jsExpressionable|null
+     * @throws Exception
+     * @throws \atk4\core\Exception
+     */
+    private function _getProperAction($response)
+    {
+        $action = null;
+        if ($response instanceof View) {
+            $action = $this->_jsRenderIntoModal($response);
+        } else if (is_string($response)) {
+            $action = new jsExpression('alert([])', [$response]);
+        } else if ($response instanceof jsExpressionable) {
+            $action = $response;
+        } else {
+            throw new Exception(['Incorrect callback. Response must be of type jsExpressionable, View, or String.', 'r' => $response]);
+        }
+
+        return $action;
+    }
+
+    /**
+     * Render View into modal.
+     *
+     * @param View $response
+     *
+     * @return jsExpression
+     * @throws \atk4\core\Exception
+     */
+    private function _jsRenderIntoModal($response)
+    {
+        $html = '<div class="ui modal"> <i class="close icon"></i>  <div class="content atk-content"> ';
```

Comment:
Quite hardcoded, is possible to render this using standard `View` components?

Reply:
in fact, why not passing Modal directly. Then developper would have total control of Modal to pass in. I will make some change here.

Reply:
Very good!